x-pack/elastic-agent/docs/elastic-agent-command-line.asciidoc

```diff
@@ -8,40 +10,276 @@
 
 beta[]
 
-The `elastic-agent run` command provides flags that alter the behavior of an
-agent:
+{agent} provides commands for running the agent and doing common tasks:
 
-`--path.home`::
-The home directory of the {agent}. `path.home` determines the location of the
-configuration files and data directory.
+* <<elastic-agent-enroll-command,enroll>>
+* <<elastic-agent-help-command,help>>
+* <<elastic-agent-inspect-command,inspect>>
+* <<elastic-agent-run-command,run>>
+* <<elastic-agent-version-command,version>>
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-enroll-command]]
+== elastic-agent enroll
+
+Enroll the {elastic-agent} in {fleet}.
+
+We recommend that you run this command as the root user because some
+integrations require root privileges to collect sensitive data. This command
+overwrites the `elastic-agent.yml` file in the agent directory.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent enroll <kibana_url> <enrollment_token> [--ca-sha256 <string>]
+                     [--certificate-authorities <string>] [--force] [--help]
+                     [--insecure][global-flags]
+----
+
+[discrete]
+=== Options
+
+`kibana_url`::
+Required. URL of the {kib} endpoint where {fleet} is running.
+
+`enrollment_token`::
+Required. Enrollment token generated by {fleet}. You can use the same
+enrollment token for multiple agents.
+
+`--ca-sha256 <string>`::
+Comma-separated list of certificate authority hash pins used for certificate
+verification.
+
+`--certificate-authorities <string>`::
+Comma-separated list of root certificates used for server verification.
+
+`--force`::
+Force overwrite of current configuration without prompting for confirmation.
+This flag is helpful when using automation software or scripted deployments.
+
+`--help`::
+Show help for the `enroll` command.
+
+`--insecure`::
+Allow an insecure connection to {kib}. We strongly recommend that you use a
+secure connection.
+
+{global-flags-link}
+
+[discrete]
+=== Example
+
+[source,shell]
+----
+elastic-agent enroll http://localhost:5601 ZnmNIdzVITUJua2QIdU5FTWROVjY6dHY2N1EybWNTMUdPejg5ODbYcVpNUQ==
+----
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-help-command]]
```

Comment:
Is this config part generated? If manual, it seems like it is very easy to get out of sync.

Reply:
Yah, I agree. It's manual right now. We need to find a way to generate this content, but I think the layout needs to be similar to how we document command reference info across the library. We aren't consistent at all, plus we don't publish our syntax conventions (some docs use angle brackets for variables, some use uppercase...it's a mess). This isn't good for our users. I might be in the minority because I like the old railroad style syntax diagrams where you can see exactly what's required, optional, and default just by looking at the diagram, and you can quickly drill down into descriptions of each parameter. :-) But I think modern users aren't accustomed to those diagrams, and most CLIs aren't so complicated that you need them.

So I tried to do something similar to how `git` documents their commands. I also looked at the ES docs. They provide a bit more syntax guidance around either/or, but I left that out because I wanted to keep things simple. I can add that, if we want.

We could decide not to document the commands, but then users wouldn't be able to discover them through a google search. 

So all this was going through my mind when I worked on these docs, but had to focus on just getting them updated and better for the beta. I'll create an issue to track this discussion because I think it's important and keeps getting kicked down the road.

Reply:
@ruflin ^^

Reply:
Created issue here: https://github.com/elastic/integrations-dev/issues/165